### PR TITLE
Feature/translation service backend

### PR DESCRIPTION
**Pull request by AssahBismarkabah**
added README file that describes the APIs for the Translation Service backend, including examples and diagrams. The section includes a description of the Twilio SMS Gateway API, the Translation Service API, and the PowerPay Service API, as well as a table summarizing the operations on the API endpoints.

### CODE REVIEW

**Comment by AssahBismarkabah** [comment]
hello sir  @stephane-segning kindly review this PR

File: power-pay-translator/README.md [hunk 1]
```diff
@@ -1,3 +1,252 @@
...
+
+ A custom API for processing messages and performing operations in the PowerPay system.
+### Cloud-native database:
+ A scalable and highly available database for storing system data.
+### Cloud platform:
+
+ A cloud platform, such as GCP or AWS, for deploying the Translation Service backend.
+
+
+# Architecture <a name="architecture"></a>
```
Comment (stephane-segning) [inline power-pay-translator/README.md:89]: Also the Translation Service Backend:
- receive requests from the P.P.Service to send someone an SMS;
- this is the way we 
  - respond to user that send an SMS, or 
  - send someone OTP for login

File: power-pay-translator/README.md [hunk 2]
```diff
@@ -1,3 +1,252 @@
...
+ The Translation Service backend receives messages from the Twilio SMS gateway using the Twilio API.
+
+### PowerPay Service:
+
+ The Translation Service backend forwards messages to the PowerPay Service using the PowerPay API.
+### Database:
+
+ The Translation Service backend reads and writes data to the database as needed.
+
+# Technology
```
[the remark this answers is not in the document]
Comment (stephane-segning) [reply power-pay-translator/README.md:69]: Even the logging tech for the rust app,...

File: power-pay-translator/README.md [hunk 3]
```diff
@@ -1,3 +1,252 @@
...
+PowerPay Service: The Translation Service forwards messages to the PowerPay Service using the PowerPay API. The PowerPay Service processes the message and performs the necessary operations, such as sending money, registering users, and checking balances. Once the operation is complete, the PowerPay Service sends a response back to the Translation Service, which then forwards the response to the Twilio SMS gateway to be sent back to the user as an SMS message.
+Database: The Translation Service and PowerPay Service both read and write data to the database as needed. The database stores all the necessary information for the system to operate, such as user data, transaction history, and account balances. The Translation Service and PowerPay Service use the database to persist data and to query for information that is needed to process messages and perform operations.
+
+These interactions are illustrated in the following diagram:
+
+![alt text](image.png)
```
Comment (stephane-segning) [inline power-pay-translator/README.md:149]: Please do your graphs and illustrations using [mermaid flowchart](https://mermaid.js.org/syntax/flowchart.html)

File: power-pay-translator/README.md [hunk 4]
```diff
@@ -1,3 +1,252 @@
...
+# Defining between component API
+
+Twilio SMS Gateway API
+
+The Twilio SMS Gateway API is a third-party API that is used to send and receive SMS messages. The Translation Service backend uses the Twilio API to receive messages from the Twilio SMS Gateway.
+
+Here is an example of how to send an SMS message using the Twilio API, you can refare this url for more information on the implementation. 
+https://www.twilio.com/docs/glossary/what-is-sms-api-short-messaging-service
+
+````js
```
Comment (stephane-segning) [inline power-pay-translator/README.md:160]: This code is using https://crates.io/crates/twilio. Did you check it is correct?

Comment (stephane-segning) [reply power-pay-translator/README.md:160]: If yes, nice. But remove it anyway. Just put the name of the lib used

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Good work overall @AssahBismarkabah! But you still need to refine some points. For starter, please remove all code blocks, as these might change. The architecture is good enough, so capitalize on that one.

**Comment by stephane-segning** [comment]
@AssahBismarkabah you think it'll be better to have two entry point Endpoints on the translation service, named `/messages` and `/transfers`? Remember that twilio is calling your app only on one place.

**Comment by stephane-segning** [comment]
Wouldn't it be better to have one, which will then translate into the multitude for calling the P.P.Service? That way another endpoint could just be the `/health`, for checking on the server

**Comment by AssahBismarkabah** [comment]
ok sir @stephane-segning   i will do the changes now

File: power-pay-translator/README.md [hunk 5]
```diff
@@ -1,3 +1,252 @@
...
+
+ A third-party service that sends SMS messages to the Translation Service backend.
+
+### Translation Service:
+
+ A Rust-based microservice that receives messages from the Twilio SMS gateway, parses and validates the messages, and forwards them to the PowerPay Service.
+### PowerPay Service:
+
+ A microservice that processes the messages and performs the necessary operations, such as sending money, registering users, and checking balances.
+### Database:
```
[the remark this answers is not in the document]
Comment (AssahBismarkabah) [reply power-pay-translator/README.md:51]: ok sir @stephane-segning  i get the reason now

**Comment by AssahBismarkabah** [comment]
hello sir @stephane-segning  for the  entry point Endpoints on the translation service i named it  /messages  the other endpoints are the proposed endpoint for the power powerpay service

**Comment by AssahBismarkabah** [comment]
hello sir @stephane-segning did the recommended modifications and added the sequence diagram involving the end poits

**Comment by stephane-segning** [comment]
@AssahBismarkabah good work. Please put your changes in a `.md` file under the doc folder

File: power-pay-translator/README.md [hunk 6]
```diff
@@ -1,3 +1,210 @@
...
+The Translation Service API is a custom API that is used to receive messages from the SMS Gateway, parse and validate the messages, and forward them to the PowerPay Service. The Translation Service API is implemented as a Rust-based microservice.
+
+
+PowerPay Service API
+The PowerPay Service API is a custom API that is used to process messages and perform the necessary operations, such as sending money, registering users, and checking balances. The PowerPay Service API is implemented as a microservice.
+
+
+The Table below summarizes the operations on the API endpoints for the Translation Service backend:
+
+| Endpoint | Method |Description |
```
Comment (stephane-segning) [inline power-pay-translator/README.md:194]: You're making a confusion here. `/handle_messages` and `/send_sms` are present on the Translator service (TS). The two others are on the PPService and the TS is calling that

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Good work. Please proceed to the changes

**Comment by stephane-segning** [comment]
You're making a confusion here. /handle_messages and /send_sms are present on the Translator service (TS). The two others are on the PPService and the TS is calling that

**Comment by AssahBismarkabah** [comment]
hello sir @stephane-segning did the necessary modifications sir.

**Comment by stephane-segning** [comment]
Awesome work!

**Pull request merged**